### PR TITLE
Multiasset autoindex

### PR DESCRIPTION
**Pull request by steven2308**
# Description

Implements MultiAsset AutoIndex extension. Recreates docs and fixes bug on MultiAsset hooks. Some where calling using `uint256` for assetId instead of `uint64`

# Checklist

- [x] Verified code additions
- [x] Updated NatSpec comments (if applicable)
- [x] Regenerated docs
- [x] Ran prettier
- [x] Added tests to fully cover the changes

### CODE REVIEW

File: contracts/RMRK/extension/multiAssetAutoIndex/IRMRKMultiAssetAutoIndex.sol [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.16;
+
+import "../../multiasset/IRMRKMultiAsset.sol";
+
+interface IRMRKMultiAssetAutoIndex is IRMRKMultiAsset {
+    /**
+     * @notice Accepts an asset at from the pending array of given token.
```
Comment (ThunderDeliverer) [inline contracts/RMRK/extension/multiAssetAutoIndex/IRMRKMultiAssetAutoIndex.sol:9]:
```suggestion
     * @notice Accepts an asset from the pending array of given token.
```

File: contracts/RMRK/extension/multiAssetAutoIndex/IRMRKMultiAssetAutoIndex.sol [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+     * @dev Migrates the asset from the token's pending asset array to the token's active asset array.
+     * @dev Active assets cannot be removed by anyone, but can be replaced by a new asset.
```
Comment (ThunderDeliverer) [inline contracts/RMRK/extension/multiAssetAutoIndex/IRMRKMultiAssetAutoIndex.sol:11]:
```suggestion
     * @dev An active asset cannot be removed by anyone, but can be replaced by a new asset.
```

File: contracts/RMRK/extension/multiAssetAutoIndex/RMRKMultiAssetAutoIndex.sol [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+contract RMRKMultiAssetAutoIndex is IRMRKMultiAssetAutoIndex, RMRKMultiAsset {
+    // Mapping of tokenId to assetId to index on the _pendingAssetIndex array
+    mapping(uint256 => mapping(uint256 => uint256)) private _pendingAssetIndex;
+
+    constructor(
+        string memory name_,
+        string memory symbol_
+    ) RMRKMultiAsset(name_, symbol_) {}
+
+    function supportsInterface(
```
Comment (ThunderDeliverer) [inline contracts/RMRK/extension/multiAssetAutoIndex/RMRKMultiAssetAutoIndex.sol:17]:
```suggestion
    /// @inheritdoc IERC165
    function supportsInterface(
```

File: contracts/RMRK/extension/multiAssetAutoIndex/RMRKMultiAssetAutoIndex.sol [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    /**
+     * @inheritdoc IRMRKMultiAssetAutoIndex
+     */
+    function rejectAsset(uint256 tokenId, uint64 assetId) public {
+        _rejectAsset(tokenId, _getIndex(tokenId, assetId), assetId);
+    }
+
+    function _getIndex(
```
Comment (ThunderDeliverer) [inline contracts/RMRK/extension/multiAssetAutoIndex/RMRKMultiAssetAutoIndex.sol:39]: Could you add documentation for this method?

File: contracts/RMRK/extension/multiAssetAutoIndex/RMRKMultiAssetAutoIndex.sol [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+     */
+    function _afterRejectAsset(
+        uint256 tokenId,
+        uint256 index,
+        uint64 assetId
+    ) internal override {
+        _removePendingIndex(tokenId, index, assetId);
+    }
+
+    function _removePendingIndex(
```
Comment (ThunderDeliverer) [inline contracts/RMRK/extension/multiAssetAutoIndex/RMRKMultiAssetAutoIndex.sol:68]: Could you add documentation for this method?

**Review by ThunderDeliverer — APPROVED** [review]
Love it! Just a few comment related changes 👌

**Comment by steven2308** [comment]
I've added the missing docstrings and few more I found

**Pull request merged**